### DISCUSSION TITLE
Volumetric Mesh Generation Using CGAL plugin

### DISCUSSION THREAD

**Discussion by andersonnardin** [SofaDefrost/SoftRobots#221] [Help - Building / Configuring]
Hello all,

Should the procedure described in https://project.inria.fr/softrobot/documentation/volumetric-mesh-generation-using-cgal-plugin/ work on the recent versions of SOFA (e.g. 22.12)?

I tried, but I could not find the required plugin for CGAL, resulting:
[ERROR]   [RequiredPlugin(RequiredPlugin)] Plugin not found: "CGALPlugin"

I also tried the examples in plugins\SofaPython3\lib\python3\site-packages\splib\meshing from version 21.06 (not present in 22.12). However, the error is presented:
[ERROR]   [SofaPython3::SceneLoader] Unable to completely load the scene from file '.../plugins/SofaPython3/lib/python3/site-packages/splib/meshing/cgal_sofa.py'.
Python exception:
  Exception: SPLIB is now relocated at: https://github.com/SofaDefrost/STLIB, please clone and install the plugin to use it

Could you explain whether this procedure is still available and suggest a working example?
Thank you in advance!

**Comment by EulalieCoevoet** [comment]
Hello, 

You need to add the plugin to SOFA first.

If you compile SOFA, there's a cmake option to fetch the git repository, otherwise you can find it here https://github.com/sofa-framework/CGALPlugin. 
If you're using binaries I think you can try to add the release of the plugin to your project https://github.com/sofa-framework/CGALPlugin/releases. I've never done that myself, maybe @hugtalbot could help you with that if needed?

You will find examples in the plugin directory, and I think the doc you mentioned is still up to date.

**Comment by andersonnardin** [comment]
Hi @EulalieCoevoet ,

I am using the binaries of version 22.12. I tried to add the plugin downloaded from the link you suggested, but I've got the  error:

Plugin loading failed(.../CCGALPlugin.dll): The specified module could not be found.

Based on what you said, recompiling SOFA is the only way of adding such a plugin. Is it correct?
